DSP_API/ThumbDV/thumbDV.c

```diff
@@ -745,50 +778,28 @@ static void * _thumbDV_readThread( void * param )
     DWORD event_dword;
 
     FT_STATUS status = FT_OK;
-    FT_HANDLE handle = *( FT_HANDLE * )param;
-    EVENT_HANDLE event_handle;
+    FT_HANDLE handle = *(FT_HANDLE *) param;
 
     prctl(PR_SET_NAME, "DV-Read");
 
-    pthread_mutex_init(&event_handle.eMutex, NULL);
-    pthread_cond_init(&event_handle.eCondVar, NULL);
-
     while ( !_readThreadAbort )
     {
-        // Setup RX or Status change event notification
-        status = FT_SetEventNotification(handle, FT_EVENT_RXCHAR , (PVOID)&event_handle);
-
-        struct timespec timeout;
-        clock_gettime(CLOCK_REALTIME, &timeout);
-
-        timeout.tv_sec += 2; // 2 second timeout
-
-        // Will block until
-        pthread_mutex_lock(&event_handle.eMutex);
-        pthread_cond_timedwait(&event_handle.eCondVar, &event_handle.eMutex, &timeout);
-        pthread_mutex_unlock(&event_handle.eMutex);
-
-        rx_bytes = 0;
-        status = FT_GetStatus(handle, &rx_bytes, &tx_bytes, &event_dword);
-
-        if ( status != FT_OK )
-        {
-            fprintf( stderr, "ThumbDV: error from status, status=%d\n", status );
-
-            /* Set invalid FD in sched_waveform so we don't call write functions */
-            handle = NULL;
-            sched_waveform_setHandle(&handle);
-            /* This function hangs until a new connection is made */
-            _connectSerial( &handle );
-            /* Update the sched_waveform to new valid serial */
-            sched_waveform_setHandle( &handle );
-        }
-        else if ( rx_bytes >= AMBE3000_HEADER_LEN )
-        {
-            ret = thumbDV_processSerial( handle );
-        }
-
+        sem_wait(&_read_sem);
 
+        ret = thumbDV_processSerial(handle);
+        //TODO Handle reconnection
```

Comment:
Does this need to be handled before we release?

Reply:
Does it? lol

Reply:
I don't think I know enough about this to know.  I only commented because it is a new TODO.

Reply:
I was able to implement it